redash/models/__init__.py

```diff
@@ -389,6 +389,8 @@ def groups(self):
 def should_schedule_next(previous_iteration, now, interval, time=None, day_of_week=None, failures=0):
     # if time exists then interval > 23 hours (82800s)
     # if day_of_week exists then interval > 6 days (518400s)
+    if previous_iteration is None:
+        return False
```

Comment:
Why not return True here instead of the `or not retrieved_at` in the calling function?

Reply:
I don't remember there being a specific reason. I think both ways are logically equivalent.

Reply:
While it's true that both will have the same result, it does make the code more confusing: `should_schedule_next` returns `False`, but we override it.. it will be more readable to have it return True which is what we want in this case anyway.